### PR TITLE
[risk=low][RW-6300] transition from getCdrVersions() to getCdrVersionsByTier()

### PR DESCRIPTION
**Pull request by jmthibault79**
Description:

This does not meet the AC for RW-6300.  It's a necessary step to get there, however.

The original endpoint `getCdrVersions()` returns a `CdrVersionListResponse` with information about the Registered Tier CDR Versions.  A new endpoint `getCdrVersionsByTier()` returns a `CdrVersionTiersResponse` with information about the CDR Versions in all tiers (Registered and Controlled or just Registered, depending on environment).  This PR replaces the use of the former endpoint/entity in favor of the newer versions.  **No visual or behavior changes are expected.  If you find any, that's a bug.**  This PR also does not enable access to any Controlled Tier CDR Version - and one does exist in the Test and Local environments.  **If you are able to access any, that's a bug.**

Local manual testing - anything relating to CDR Versions: create, edit, duplicate workspaces, changing CDR Version where possible (but not tier).  create notebook runtime.  copy resources between workspaces across CDR Versions (but not across tiers).  Observe CDR Version displays in Recent Resources, Workspace About, Workspace Navbar.

---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [x] This PR includes appropriate unit tests
- [x] I have run and tested this change locally
- [ ] I have run the E2E tests on ths change against my local UI and/or API server with `yarn test-local` or [yarn test-local-devup](https://github.com/all-of-us/workbench/blob/master/e2e/README.md#examples)
- [x] If this includes a UI change, I have taken screen recordings or screenshots of the new behavior and notified the PO and UX designer
- [x] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [x] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

File: ui/src/app/cohort-search/list-search/list-search.component.tsx [hunk 1]
```diff
@@ -284,9 +285,8 @@ export const ListSearch = fp.flow(withCdrVersions(), withCurrentWorkspace(), wit
       };
     }
     componentDidMount(): void {
-      const {cdrVersionListResponse, searchTerms, searchContext: {source}, workspace: {cdrVersionId}} = this.props;
-      const cdrVersions = cdrVersionListResponse.items;
-      this.setState({cdrVersion: cdrVersions.find(cdr => cdr.cdrVersionId === cdrVersionId)});
```
Comment (jmthibault79) [inline ui/src/app/cohort-search/list-search/list-search.component.tsx:289]: this `find` is a little more complicated with the new response structure so I moved it into a utils function instead of copying the new logic everywhere

File: ui/src/app/pages/signed-in/component.ts [hunk 2]
```diff
@@ -95,7 +95,7 @@ export class SignedInComponent implements OnInit, OnDestroy, AfterViewInit {
         this.profile = profile as unknown as FetchProfile;
         setInstitutionCategoryState(this.profile.verifiedInstitutionalAffiliation);
         if (hasRegisteredAccess(this.profile.accessTierShortNames)) {
-          cdrVersionsApi().getCdrVersions().then(resp => {
+          cdrVersionsApi().getCdrVersionsByTier().then(resp => {
```
Comment (jmthibault79) [inline ui/src/app/pages/signed-in/component.ts:98]: This is the change that drives everything else: this call returns a `CdrVersionTiersResponse` which includes Controlled Tier instead of `CdrVersionListResponse` which does not.

File: ui/src/app/pages/workspace/workspace-edit.tsx [hunk 3]
```diff
@@ -185,6 +191,9 @@ export const styles = reactStyles({
 
 const CREATE_BILLING_ACCOUNT_OPTION_VALUE = 'CREATE_BILLING_ACCOUNT_OPTION';
 
+// default to creating workspaces in the Registered Tier
```
Comment (jmthibault79) [inline ui/src/app/pages/workspace/workspace-edit.tsx:194]: Note: Registered is still the ONLY tier accessible from the UI after this change.  This PR is only part of the journey.

**Comment by jmthibault79** [comment]
I'm marking this DON'T MERGE until after the release, out of an abundance of caution

File: ui/src/app/pages/login/sign-in.tsx [hunk 4]
```diff
@@ -107,7 +107,6 @@ export const StepToImageConfig: Map<SignInStep, BackgroundImageConfig> = new Map
   }]]
 );
 
-
```
Comment (calbach) [inline ui/src/app/pages/login/sign-in.tsx:110]: nit: revert this and profile-page.spec.tsx? appears to be unrelated

File: ui/src/app/utils/cdr-versions.spec.tsx [hunk 5]
```diff
@@ -1,47 +1,90 @@
 import * as React from "react";
 
-import {Workspace, CdrVersion, CdrVersionListResponse} from 'generated/fetch';
+import {Workspace} from 'generated/fetch';
 import {WorkspaceStubVariables} from 'testing/stubs/workspaces';
-import {cdrVersionListResponse} from 'testing/stubs/cdr-versions-api-stub';
-import {getCdrVersion, getDefaultCdrVersion, hasDefaultCdrVersion} from "./cdr-versions";
+import {cdrVersionTiersResponse} from 'testing/stubs/cdr-versions-api-stub';
+import {getCdrVersion, hasDefaultCdrVersion, getDefaultCdrVersionForTier} from "./cdr-versions";
```
Comment (calbach) [inline ui/src/app/utils/cdr-versions.spec.tsx:6]: nit: partially an existing issue, but please qualify these from the project root

File: ui/src/app/utils/cdr-versions.spec.tsx [hunk 6]
```diff
@@ -1,47 +1,90 @@
...
+import {AccessTierShortNames} from "./access-tiers";
 
+const registeredCdrVersionTier = cdrVersionTiersResponse.tiers[0];
```
Comment (calbach) [inline ui/src/app/utils/cdr-versions.spec.tsx:9]: I suggest moving these definitions into `cdr-versions-api-stub` instead so the data definitions live in the same place, or else switching this to do a lookup by tier name - whatever lets you avoid a non-obvious dependency on this array order

File: ui/src/app/utils/cdr-versions.spec.tsx [hunk 7]
```diff
@@ -1,47 +1,90 @@
...
 
 const stubWorkspace: Workspace = {
     name: WorkspaceStubVariables.DEFAULT_WORKSPACE_NAME,
     id: WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
     namespace: WorkspaceStubVariables.DEFAULT_WORKSPACE_NS
 };
 
-const defaultCdrVersion = cdrVersionListResponse.items[0];
-const altCdrVersion = cdrVersionListResponse.items[1];
+const defaultCdrVersion = registeredCdrVersionTier.versions[0];
```
Comment (calbach) [inline ui/src/app/utils/cdr-versions.spec.tsx:18]: Existing issue, but same comment as above

File: ui/src/app/utils/cdr-versions.tsx [hunk 8]
```diff
@@ -1,19 +1,39 @@
...
-function getCdrVersion(workspace: Workspace, cdrList: CdrVersionListResponse): CdrVersion {
-  return cdrList.items.find(v => v.cdrVersionId === workspace.cdrVersionId);
+function getCdrVersionTier(accessTierShortName: string, cdrTiers: CdrVersionTiersResponse): CdrVersionTier {
+  return cdrTiers.tiers.find(v => v.accessTierShortName === accessTierShortName);
 }
 
-function getDefaultCdrVersion(cdrList: CdrVersionListResponse): CdrVersion {
-  return cdrList.items.find(v => v.cdrVersionId === cdrList.defaultCdrVersionId);
+function getCdrVersion(workspace: Workspace, cdrTiers: CdrVersionTiersResponse): CdrVersion {
+  const tier = getCdrVersionTier(workspace.accessTierShortName, cdrTiers);
```
Comment (calbach) [inline ui/src/app/utils/cdr-versions.tsx:9]: nit: this method could arguably be more simply implemented as `findCdrVersion(workspace.cdrVersion)`. It's not strictly necessary to check the tier when doing this lookup.

Alternatively, this may also be simple enough that callers can just use that instead of having a helper.

File: ui/src/testing/stubs/cdr-versions-api-stub.ts [hunk 9]
```diff
@@ -1,53 +1,72 @@
...
   static ALT_WORKSPACE_CDR_VERSION = 'Alternative CDR Version';
   static ALT_WORKSPACE_CDR_VERSION_ID = 'altCdrVersion';
+  static CONTROLLED_TIER_CDR_VERSION = 'Controlled Tier CDR Version';
+  static CONTROLLED_TIER_CDR_VERSION_ID = 'ctCdrVersion';
 }
 
-export const cdrVersionListResponse: CdrVersionListResponse = {
-  defaultCdrVersionId: CdrVersionsStubVariables.DEFAULT_WORKSPACE_CDR_VERSION_ID,
-  items: [
+export const cdrVersionTiersResponse: CdrVersionTiersResponse = {
```
Comment (calbach) [inline ui/src/testing/stubs/cdr-versions-api-stub.ts:18]: opt: existing issue, but I would generally define test data like this with a function instead of a static variable. With the current approach, all it takes is one test method to modify this data, and then it's tainted for all subsequent runs, which can be tricky to track down. (Note: I may have been the one to originally define this as a static variable)

One idea: perhaps this is a good use case for the `get` keyword, which conveniently contains the change to just this file as well.

**Review by calbach — APPROVED** [review]
Thanks, looks good - just nitpicks.

File: ui/src/testing/stubs/cdr-versions-api-stub.ts [hunk 10]
```diff
@@ -1,53 +1,90 @@
...
-        items: this.cdrVersions,
-      });
+  getCdrVersionsByTier(options?: any): Promise<CdrVersionTiersResponse> {
+    return new Promise<CdrVersionTiersResponse>(resolve => {
+      resolve(cdrVersionTiersResponse);
     });
   }
 }
+
+export {
```
Comment (petesantos) [inline ui/src/testing/stubs/cdr-versions-api-stub.ts:81]: Why the change to this style of export?

**Review by petesantos — APPROVED** [review]
Looks good

File: ui/src/testing/stubs/cdr-versions-api-stub.ts (hunk 10 above)
Comment (jmthibault79) [reply ui/src/testing/stubs/cdr-versions-api-stub.ts:81]: Going off the advice of style-guide.md

Comment (petesantos) [reply ui/src/testing/stubs/cdr-versions-api-stub.ts:81]: I didn't realize that. A quick scan of the code seems to show that this style is not heavily used, but it is used in some places.

What is your preference for coding style?

I prefer exporting where the function is defined. When I read the code I find having that information close together is useful, otherwise I would have to navigate to the bottom of the file or to see if the function is meant to be used externally. I don't recall ever being in a situation where I need to scan the file to see what all the exports are - for me, that is not the way I read code. Having the breadth of all exports for a given file in one place isn't something I am usually after, but I could grep it out of the file if needed. I think it would be harder to do the reverse operation (put the exports close).

This is a bit out of scope for this PR. But I would like to get the general feel / opinions of the AoU team around this.

Comment (jmthibault79) [reply ui/src/testing/stubs/cdr-versions-api-stub.ts:81]: I agree that it would be good to get a team consensus for style questions like these.

I don't have a real opinion on this particular point.

Comment (calbach) [reply ui/src/testing/stubs/cdr-versions-api-stub.ts:81]: Seems like that initial style guide was checked in with minimal review, with the intent that it was capturing our "effective" codebase style at the time - so asserting this specific export style was probably an overreach, or just too localized. I also have a preference to just export with the declaration (and this is what I've generally done in practice).

I definitely support switching the style guidance if you want to propose the change (ideally we could also switch the codebase to be consistent).

File: ui/src/testing/stubs/cdr-versions-api-stub.ts (hunk 9 above)
Comment (calbach) [reply ui/src/testing/stubs/cdr-versions-api-stub.ts]: Note: seems you actually can't define a getter at the top level without doing some shenanigans, so ignore my idea.
